Content.Server/_DV/Abilities/Chitinid/ChitinidSystem.cs

```diff
@@ -0,0 +1,123 @@
+    [Dependency] private readonly DamageableSystem _damageable = default!;
+    [Dependency] private readonly MobStateSystem _mobState = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<ChitinidComponent, ComponentInit>(OnInit);
+        SubscribeLocalEvent<ChitinidComponent,ChitziteActionEvent>(OnChitzite);
+        SubscribeLocalEvent<ChitinidComponent,MapInitEvent>(OnMapInit);
+    }
+
+    private Queue<EntityUid> RemQueue = new();
+
+
+    public override void Update(float frameTime)
+    {
+        base.Update(frameTime);
+        foreach (var bug in RemQueue)
+        {
+            RemComp<CoughingUpChitziteComponent>(bug);
+        }
+        RemQueue.Clear();
+
+        var query = EntityQueryEnumerator<ChitinidComponent, DamageableComponent>();
+        while (query.MoveNext(out var uid, out var chitinid, out var damageable))
+        {
+            if (_gameTiming.CurTime < chitinid.NextUpdate)
+                continue;
+
+            chitinid.NextUpdate += chitinid.UpdateInterval;
+
+            if (chitinid.AmountAbsorbed >= chitinid.MaximumAbsorbed && !_mobState.IsDead(uid))
+                continue;
+
+            var damage = new DamageSpecifier(_prototypeManager.Index<DamageTypePrototype>("Radiation"), -0.5);
+            var delta = _damageable.TryChangeDamage(uid, damage);
+            if (delta != null)
+            {
+                chitinid.AmountAbsorbed += -(delta.GetTotal().Float());
+                if (chitinid.ChitziteAction != null && chitinid.AmountAbsorbed >= chitinid.MaximumAbsorbed)
+                {
+                    _actionsSystem.SetCharges(chitinid.ChitziteAction, 1); // You get the charge back and that's it. Tough.
+                    _actionsSystem.SetEnabled(chitinid.ChitziteAction, true);
+                }
+            }
+        }
+
+        foreach (var (chitziteComponent, chitinidComponent) in EntityQuery<CoughingUpChitziteComponent, ChitinidComponent>())
+        {
+            chitziteComponent.Accumulator += frameTime;
+            if (chitziteComponent.Accumulator < chitziteComponent.CoughUpTime.TotalSeconds)
+                continue;
+
+            chitziteComponent.Accumulator = 0;
+            SpawnChitzite(chitziteComponent.Owner, chitinidComponent);
+            RemQueue.Enqueue(chitziteComponent.Owner);
+        }
+    }
+
+    private void OnMapInit(Entity<ChitinidComponent> ent, ref MapInitEvent args)
+    {
+        ent.Comp.NextUpdate = _gameTiming.CurTime + ent.Comp.UpdateInterval;
+    }
+
+    private void OnInit(EntityUid uid, ChitinidComponent component, ComponentInit args)
+    {
+        if (component.ChitziteAction != null)
+            return;
+
+        _actionsSystem.AddAction(uid, ref component.ChitziteAction, component.ChitziteActionId);
+    }
+
+    private void OnChitzite(EntityUid uid, ChitinidComponent component, ChitziteActionEvent args)
```

Comment:
`Entity<ChitinidComponent> ent, ref ChitziteActionEvent args` and changes stuff to use ent.Comp instead of component accordingly